### PR TITLE
Create a dry run mode

### PR DESCRIPTION
**Pull request by LyonesGamer**
Depends on #8.

Creates a dry run mode which prints out the actions to be taken but does not create or update anything.

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -388,7 +392,11 @@ func updateIssue(ghIssue github.Issue, jIssue jira.Issue, ghClient github.Client
 				log.Infof("  Summary: %s", fields.Summary)
 			}
 			if fields.Description != "" {
-				log.Infof("  Description: %s...", fields.Description[0:20])
+				if len(fields.Description) > 20 {
```
Comment (squat) [inline cmd/root.go]: These changes seem unrelated to the dry-run option. Separate this into it's own PR so we can track and revert dry-run changes separately from this bug fix.

File: cmd/root.go [hunk 2]
```diff
@@ -357,22 +361,44 @@ func updateIssue(ghIssue github.Issue, jIssue jira.Issue, ghClient github.Client
...
-		if err != nil {
-			log.Errorf("Error updating JIRA issue %s: %s", jIssue.Key, err)
-			defer res.Body.Close()
-			body, _ := ioutil.ReadAll(res.Body)
-			log.Debugf("Error body: %s", body)
-			return err
+			if err != nil {
+				log.Errorf("Error updating JIRA issue %s: %s", jIssue.Key, err)
+				defer res.Body.Close()
+				body, _ := ioutil.ReadAll(res.Body)
```
Comment (squat) [inline cmd/root.go]: handle this error

File: cmd/root.go [hunk 3]
```diff
@@ -357,22 +361,44 @@ func updateIssue(ghIssue github.Issue, jIssue jira.Issue, ghClient github.Client
...
+		}
 
 		issue := &jira.Issue{
 			Fields: &fields,
 			Key:    jIssue.Key,
 			ID:     jIssue.ID,
 		}
 
-		issue, res, err := jClient.Issue.Update(issue)
+		if !dryRun {
```
Comment (squat) [inline cmd/root.go]: this will not scale well and will get unmanageable quite fast. the behavior of this function is now being drastically altered by a variable that is not a param. Testing this is messy and ensuring correctness becomes increasingly difficult to reason about. The correct way to do this is for all of these functions to be methods on a issue-sync config struct that encapsulates all the configuration, including dry-run. These methods then operate on a method receiver that holds all the information needed to control the operation of the function. Either that, or all of these methods take an additional parameter `dryRun`.

**Comment by LyonesGamer** [comment]
This functionality is now replicated in #15. We may be able to simply close this PR. Let me check...

File: cmd/root.go [hunk 4]
```diff
@@ -383,19 +391,46 @@ func updateIssue(ghIssue github.Issue, jIssue jira.Issue, ghClient github.Client
...
+		} else {
+			log.Info("")
+			log.Infof("Update JIRA issue %s with GitHub issue #%d:", jIssue.Key, ghIssue.GetNumber())
+			if fields.Summary != jIssue.Fields.Summary {
+				log.Infof("  Summary: %s", fields.Summary)
+			}
+			if fields.Description != "" {
+				fields.Description = newlineReplaceRegex.ReplaceAllString(fields.Description, "\\n")
+				if len(fields.Description) > 20 {
+					log.Infof("  Description: %s...", fields.Description[0:20])
```
Comment (squat) [inline cmd/root.go]: why do some of these have leading whitespace?

File: cmd/root.go [hunk 5]
```diff
@@ -559,17 +619,32 @@ func updateComment(ghComment github.IssueComment, jComment jira.Comment, jIssue
...
-	// TODO: Abstract this into its own method
-	req, err := jClient.NewRequest("PUT", fmt.Sprintf("rest/api/2/issue/%s/comment/%s", jIssue.Key, jComment.ID), request)
-	if err != nil {
-		log.Errorf("Error creating comment update request: %v", err)
-		return err
-	}
+	if !dryRun {
+		req, err := jClient.NewRequest("PUT", fmt.Sprintf("rest/api/2/issue/%s/comment/%s", jIssue.Key, jComment.ID), request)
+		if err != nil {
+			log.Errorf("Error creating comment update request: %s", err)
```
Comment (squat) [inline cmd/root.go]: use `%v`

File: cmd/root.go [hunk 6]
```diff
@@ -39,6 +39,8 @@ var (
 	isLastUpdateFieldID string    // The customfield ID of the Last Issue-Sync Update field in JIRA
 
 	project jira.Project
+
+	dryRun bool
```
Comment (squat) [inline cmd/root.go]: this should be a field on a config struct, not a global variable.

File: cmd/root.go (hunk 4 above)
Comment (LyonesGamer) [inline cmd/root.go]: The fields all have whitespace for clarity. All the dry-runs print:

Doing action
  Field1: ...
  Field2: ...

**Comment by LyonesGamer** [comment]
> I realize that is coming in #15, but that work should really just have been written in this PR.

Okay. I just felt that that would be too much all in one PR, especially since it depended on the config struct as well, which is also its own big changes.
